src/iperf.h

```diff
@@ -294,7 +294,7 @@ struct iperf_test
     int       bind_port;                        /* --cport option */
     int       server_port;
     int       omit;                             /* duration of omit period (-O flag) */
-    int       duration;                         /* total duration of test (-t flag) */
+    uint32_t  duration;                         /* total duration of test (-t flag) */
```

Comment:
I think that `duration` can and should remain an `int`, right? You check in line +1316 below should make this value safe...in fact if you change `duration` to be unsigned, then `test->duration < 0` below will never be true. Also the return value of `atoi()` is an `int`, not an `unsigned int`.

Reply:
Agree `duration` should be changed back to signed.  I didn't take into account that atoi() returns int ... I first changed to unsigned `duration` but than added the `test->duration < 0` check to ensure that if it will be changed back to signed, negative values will still fail.

Changed now back to signed, but to `int64_t` instead of `int` and `atol()` instead of `atoi()`.  This is because for a very large absolute value, e.g. `-99999999999999999999999`, atoi() returns 0 which is the failure value, but 0 is a valid timeout.  The change does not solve the issue, but much larger absolute values should be used to cause it.

If the above change is an overkill, and `int` / `atoi()` should be used, please let me know.

Reply:
I think I'd like to stick to `int` / `atoi()`. I'm not sure if `int64_t` and `atol()` are always comparable without conversions, and if someone needs a test to run for longer than 2^(32-1) seconds well, maybe they might need to rethink that a bit.

Reply:
O.k Changed back to `int`/ `atoi()` (with rebase).

Reply:
Thanks!